### PR TITLE
added logic for password reset

### PR DESCRIPTION
**Pull request by gnorbsl**

### CODE REVIEW

File: controllers/email.js [hunk 1]
```diff
@@ -84,44 +110,39 @@ const sendMail = async (from, to, subject, message, attachments = undefined) =>
...
-      cid: 'Img1_2x.jpg', // same cid value as in the html img src
-    },
-  ];
-
-  const body = template.replace('linkplaceholder', `${process.env.BASE_URL}/users/verify/${hash}`);
-
-  console.log('BODY');
+
+  const body = template
+    .replace('linkplaceholder', `${process.env.BASE_URL}/users/verify/${hash}`)
```
Comment (Enubia) [inline controllers/email.js:115]: intended to not use ```%```linkplaceholder```%```?

File: controllers/email.js [hunk 2]
```diff
@@ -84,44 +110,39 @@ const sendMail = async (from, to, subject, message, attachments = undefined) =>
...
-    attachments,
+    templateAttachments,
+  );
+};
+
+
+const sendPasswordResetMail = async (to, hash) => {
+
+  const body = template
+    .replace('linkplaceholder', `${process.env.BASE_URL}/users/password/reset/${hash}`)
```
Comment (Enubia) [inline controllers/email.js:134]: same here as mentioned above

File: routes/users.js [hunk 3]
```diff
@@ -387,4 +392,119 @@ router.get('/choose', redirectLogin, async (req, res) => {
...
+// @access  Private
+// @tested
+router.post('/password/request', async (req, res) => {
+  try {
+
+    if (!req.body.email) return handleError(res, 400, 'email missing');
+
+    const userObject = await User.findOne({ email: req.body.email });
+
+    if (!userObject) return handleError(res, 400, 'user not found');
```
Comment (Enubia) [inline routes/users.js:421]: no need to return the handleError, it sends a response inside of it

**Review by Enubia — CHANGES_REQUESTED** [review]
looks good so far, just a few things unlcear

File: routes/users.js (hunk 3 above)
Comment (gnorbsl) [reply routes/users.js:421]: if I dont return the rest of the code is getting called though?

**Pull request merged**